### DISCUSSION TITLE
error boot_fat.py

### DISCUSSION THREAD

**Discussion by Arie1956** [micropython/micropython#15707] [RP2040 / Pico] [answered]
Can someone tell me what is in line 12 of boot_fat.py?
I just updated the firmware om my Arduino RP2040 Connect and get the error

MPY: soft reboot
Traceback (most recent call last):
  File "_boot_fat.py", line 12, in <module>
OSError: [Errno 19] ENODEV

Have searched the internet without luck

**Comment by shariltumin** [comment, marked as answer]
This is the contents of the "_boot_fat.py" file:

```py
import vfs
import machine, rp2


# Try to mount the filesystem, and format the flash if it doesn't exist.
bdev = rp2.Flash()
try:
    fs = vfs.VfsFat(bdev)
except:
    vfs.VfsFat.mkfs(bdev)
    fs = vfs.VfsFat(bdev)
vfs.mount(fs, "/")

del vfs, bdev, fs
```

It seems that you may have a problem with your flash.

**Reply by Arie1956** [reply]
Yes I have, Found what I was doing wrong when down grading to older firmware, I had to do a factory reset using Blink.ino.elf.uf2. then reload the new firmware. version v1.23.0 has a bug but v1.22.0 works OK..
